packages/components/src/side_scroller/controls.jsx

```diff
@@ -0,0 +1,77 @@
+import clsx from 'clsx';
+import PropTypes from 'prop-types';
+import styles from './controls.module.scss';
+import { getAnimationPosition } from './utils';
+import { SHIFT_PERCENT, SHIFT_TOLERANCE, ANIMATION_DURATION, ANIMATION_FPS } from './constants';
+
+const Controls = ({
+  canScrollLeft,
+  canScrollRight,
+  node,
+  containerWidthRef,
+  contentWidthRef,
+  onAnimateScroll,
+}) => {
+  const startScrollAnimation = (target) => {
+    let time = 0;
+
+    const animateScroll = () => {
+      const newValue = getAnimationPosition(node.scrollLeft, target, time, ANIMATION_DURATION);
+
+      time += 1000 / ANIMATION_FPS;
+      node.scrollLeft = newValue;
+
+      if (newValue !== target) onAnimateScroll(animateScroll);
+    };
+
+    animateScroll();
+  };
+
+  const handleShift = (percent) => {
+    const { scrollLeft } = node;
+    const shiftAmount = Math.floor(percent * containerWidthRef.current);
+    const maxScrollPosition = contentWidthRef.current - containerWidthRef.current;
+
+    let target = scrollLeft + shiftAmount;
+    if (target + SHIFT_TOLERANCE >= maxScrollPosition) target = maxScrollPosition;
+    if (target - SHIFT_TOLERANCE <= 0) target = 0;
+
+    startScrollAnimation(target);
+  };
+
+  const handleScrollLeft = () => {
+    handleShift(-SHIFT_PERCENT);
+  };
+
+  const handleScrollRight = () => {
+    handleShift(SHIFT_PERCENT);
+  };
+
+  const renderControl = (type, handler) => (
```

Comment:
Render Control should be a component, instead of render-function. 
Though, it's so simple - I guess we can refactor it like this:

Imo you can extract it in the same file, pass `type` and `handler` as a prop, and then assign it to `leftControl`/`rightControl` variables. 

@peterhass any comments on this?